sdk/core/azure-core/src/logger.cpp

```diff
@@ -54,21 +54,26 @@ void Logger::SetLevel(Logger::Level level) { Log::SetLogLevel(level); }
 
 int Log::LoggerStringBuffer::sync()
 {
-  Log::Write(m_level, str());
+  std::string val{str()};
+  if (!val.empty())
```

Comment:
I think it is also better to move this check to `Log::Write()`

Reply:
This case exists to handle the very special case of inserting `std::endl` which calls flush, and then the `Stream` destructor calls flush again (on an empty string). I'll add a comment explaining why it's here.